src/Parser.js

```diff
@@ -58,18 +62,22 @@ class Parser {
       ext = '.' + ext;
     }
 
-    this.extensions[ext.toLowerCase()] = parser;
+    this.extensions.set(ext.toLowerCase(), parser);
   }
 
   findParser(filename) {
+    // console.log('trying to find parser for', filename);
     if (/[*+{}]/.test(filename) && glob.hasMagic(filename)) {
       return GlobAsset;
     }
 
     let extension = path.extname(filename).toLowerCase();
-    let parser = this.extensions[extension] || RawAsset;
+    // console.log({extension});
+    // console.log('registered extensions', ...this.extensions.keys());
+    let parser = this.extensions.get(extension) || RawAsset;
+    // console.log({parser});
     if (typeof parser === 'string') {
-      parser = this.extensions[extension] = require(parser);
+      parser = this.extensions.set(extension, require(parser)).get(extension);
```

Comment:
If performance is the point of this PR than this actually could be faster if you write it like this:
```Javascript
parser = require(parser);
this.extensions.set(extension, parser);
```

Test case: https://jsperf.com/assign-vs-getting-map